pytensor/tensor/math.py

```diff
@@ -343,6 +345,208 @@ def grad(self, inp, grads):
         return (g_x,)
 
 
+class TensorMax(COp):
+    """
+    Calculate the max over a given axis or over all axes.
+
+    """
+
+    nin = 2  # tensor, axis
+    nout = 1  # max val
+    E_axis = "invalid axis"
+    params_type = Generic()
+    __props__ = ("axis",)
+    _f16_ok = True
+
+    def __init__(self, axis):
+        assert isinstance(axis, tuple | list)
+        self.axis = tuple(axis)
+
+    def get_params(self, node):
+        return self.axis
+
+    def make_node(self, x):
+        x = as_tensor_variable(x)
+
+        # Keep the original shapes for axes on which we do not perform the max/argmax.
+        all_axes = set(self.axis)
+        inputs = [x]
+        out_shape = tuple(s for i, s in enumerate(x.type.shape) if i not in all_axes)
+        outputs = [
+            tensor(dtype=x.type.dtype, shape=out_shape, name="max"),
+        ]
+        return Apply(self, inputs, outputs)
+
+    def prepare_node(self, node, storage_map, compute_map, impl):
+        if len(node.inputs) == 2:
+            raise ValueError(
+                "You are trying to compile a graph with an old Argmax node.  Either reoptimize your graph or rebuild it to get the new node format."
+            )
+
+    def perform(self, node, inp, outs):
+        x = inp[0]
+        axes = self.axis
+        # max, max_idx = outs
+        (max,) = outs
+        if axes is None:
+            axes = tuple(range(x.ndim))
+        else:
+            axes = tuple(int(ax) for ax in axes)
+        max[0] = _asarray(np.max(x, axes), dtype=node.outputs[0].dtype)
+        # # Numpy does not support multiple axes for argmax
```

Comment:
some code is just commented out from the maxandargmax approach. In the final PR, they will be erased completely.